modules/javafx.controls/src/main/java/javafx/scene/control/TextInputControl.java

```diff
@@ -184,6 +173,23 @@ protected TextInputControl(final Content content) {
         getStyleClass().add("text-input");
     }
 
+    private void updateSelectedText() {
+        if (!blockSelectedTextUpdate) {
+            String txt = text.get();
+            IndexRange sel = selection.get();
+            if (txt == null || sel == null) {
+                selectedText.set("");
+            } else {
+                int start = sel.getStart();
+                int end = sel.getEnd();
+                int length = txt.length();
+                if (end > start + length) end = length;
```

Comment:
As I mentioned in [this comment in PR #73](https://github.com/openjdk/jfx/pull/73#discussion_r376528686), I think this test is wrong. The value of `start` has no impact on whether `end` is out of range. So... shouldn't this simply be `if (end > length) end = length;` ?

Reply:
You're right. The whole point of the fix was to remove the need for any clamping by preventing the "in-between" updates of the selected text property.
I've checked that the condition in line 186 is never fulfilled now anymore and removed it.
I've also removed the clamping in the line below, because the only case where start > length-1 is now that start == length in which case also end == length and txt.substring(length, length) will be just as empty as txt.substring(0, 0).